Documentation/cilium_operator.rst

```diff
@@ -0,0 +1,212 @@
+      --namespace kube-system \\
+      --set enableCnpStatusUpdates=false
+
+Heartbeat update
+^^^^^^^^^^^^^^^^
+
+The operator periodically updates the Cilium's heartbeat path key
+with the current time. The default key for this heartbeat is
+``cilium/.heartbeat`` in the KVStore. It is used by agents to validate
+that kvstore updates can be received.
+
+Policy status update
+^^^^^^^^^^^^^^^^^^^^
+
+Cilium operator performs the operation of CNP/CCNP node status updates
+when ``k8s-events-handover`` is enabled. This is done to optimize
+Kubernetes events handling in case of large clusters. For the node
+status updates to be handled by ``cilium-operator``, all the K8s events
+are mirrored to the KVstore, which are then used to perform operations
+via the operator. This operation is performed for both
+CiliumNetworkPolicy and CiliumClusterwideNetworkPolicy objects.
+
+For each CNP/CCNP object in the cluster, we start a status handler. This
+handler periodically updates the node statuses for the CNP/CCNP objects
+with the status of the policy in the corresponding node.
+
+Identity Garbage Collection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each workload in Kubernetes is assigned a security identity that is used
+for policy decision making. This identity is based on common workload
+markers like labels. Cilium supports two identity allocation mechanisms:
+
+-  CRD Identity allocation
+-  KVStore Identity allocation
+
+Both the mechanisms of identity allocation requires the
+``cilium-operator`` to perform the Garbage collection of stale
+identities. This garbage collection is necessary because a 16-bit
+unsigned integer represents the security identity, and thus we can only
+have a maximum of 65536 identities in the cluster.
+
+CRD Identity GC
+^^^^^^^^^^^^^^^
+
+CRD identity allocation uses Kubernetes custom resource
+``CiliumIdentity`` to represent a security identity. This is the default
+behaviour of Cilium and works out of the box in any K8s environment
+without any external dependency.
+
+The cilium operator maintains a local cache for CiliumIdentities with
+the last time they were seen active. A controller runs in the background
+periodically which scans this local cache and deletes identities that
+have not had their heartbeat life sign updated since
+``identity-heartbeat-timeout``.
+
+One thing to note here is that an Identity is always assumed to be live
+if it has an endpoint associated with it.
+
+KVStore Identity GC
+^^^^^^^^^^^^^^^^^^^
+
+While the CRD allocation mode for identities is more common, it is
+limited in terms of scale. When running in a very large environment, the
+more sane choice is to use the KVStore allocation mode. This mode stores
+the identities in an external store like etcd or Consul.
+
+For more information on Cilium's scalability visit :ref:`scalability_guide`.
+
+The garbage collection mechanism involves scanning the KVStore of all
+the identities using prefix(``id/``) search for identity keys. For
```

Comment:
This is not easy to read. Not sure what `using prefix(``id/``) search` refers to. Maybe rephrase?